experiments/modern/src/App.js

```diff
@@ -49,6 +22,26 @@ async function getSkin() {
   return await initialize(zip, skinXml);
 }
 
+function Container(props) {
+  const { id, children, default_x, default_y, default_visible } = props;
+  const style = {
+    position: "absolute",
+  };
+  if (default_x !== undefined) {
+    style.left = Number(default_x);
+  }
+  if (default_y !== undefined) {
+    style.top = Number(default_y);
+  }
+  if (default_visible !== undefined) {
+    style.display = default_visible ? "block" : "none";
+  }
+  return <div
+           data-node-type="container"
+           data-node-id={id}
+           style={style}>{children}</div>;
+}
```

Comment:
I wonder if we could/should duplicate the inheritance hierarchy with composition. Meaning: every component renders the component of it's parent class.

Reply:
oooh thats interesting.  Need to implement a few more components and see if the hierarchy would work that way, but seems like a good chance